### PR TITLE
Add a Makefile

### PR DESCRIPTION
**Pull request by muffix**
This adds a Makefile to the repository to make common tasks easier. It
also prepares for further refactoring of the build and test process.

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

# Description

This adds a Makefile to the repository to make common tasks easier. It
also prepares for further refactoring of the build and test process.

## Type of change

From the following, please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Please also list any relevant details for your test configuration

- [x] All targets run locally 

# Checklist:

- [x] This contribution follows the project's [code of conduct](https://github.com/bosun-monitor/bosun/blob/master/CODE_OF_CONDUCT.md)
- [x] This contribution follows the project's [contributing guidelines](https://github.com/bosun-monitor/bosun/blob/master/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,6 +24,14 @@ Please use [StackOverflow] for support questions. There exists a Slack instance
 
 Please also see also our [Code of Conduct].
 
+### Setting up your build system
+
+The repository contains a [Makefile] which should make it easy to install all dependencies. Other than that, 
+you're going to need a recent (1.13+) Go environment.
```
[the remark this answers is not in the document]
Comment (muffix) [reply CONTRIBUTING.md]: Added, thanks.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+bosun:
+	$(GOBUILD) -o $(BINARY_NAME) -v $(BOSUN_PACKAGE)
+bosun-linux:
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GOBUILD) -o $(BINARY_UNIX) -v $(BOSUN_PACKAGE)
+bosun-darwin:
+	CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GOBUILD) -o $(BINARY_MAC) -v $(BOSUN_PACKAGE)
+bosun-windows:
+	CGO_ENABLED=0 GOOS=windows GOARCH=amd64 $(GOBUILD) -o $(BINARY_WIN) -v $(BOSUN_PACKAGE)
+
+.PHONY: all
```
[the remark this answers is not in the document]
Comment (muffix) [reply Makefile:139]: Added, thanks.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+.PHONY: updatetestdeps
+updatetestdeps:
+	$(GOGET) -d -v -t -u -f ./...
+	$(GOGET) -u -v $(LINTERS)
+
+.PHONY: install
+install: deps
+	$(GOINSTALL) ./...
+
+.PHONY: golint
```
[the remark this answers is not in the document]
Comment (muffix) [reply Makefile:53]: You need to install the test dependencies with `make testdeps` first.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -77,12 +89,14 @@ We use the golang [Go code review comments] document as the basis for our style.
 Take particular note of Error Strings, Mixed Caps, and Indent Error Flow sections. 
 Also we don't have blank lines within functions.
 
+The make target `checks` will be run to
```
Comment (johnewing1) [inline CONTRIBUTING.md:92]: Missing the end of this sentence ?

**Pull request merged**